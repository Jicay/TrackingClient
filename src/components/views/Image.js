import React, {Component} from "react";
import {Link} from "react-router-dom";

export default class Image extends Component {

    constructor(props) {
        super(props);
    }

    render() {
        const {image} = this.props;

        const pathVariable = image.replace("/", "-");

        return (
            <div className='col align-content-center'>
                <Link to={'/images/' + pathVariable}>
                    <img className="img-fluid" src={'https://live.staticflickr.com/' + image + '.jpg'}/>
                </Link>
            </div>
        )
    }


}