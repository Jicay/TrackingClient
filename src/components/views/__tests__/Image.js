import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from "react-router";
import Image from "../Image";

it('renders correctly', () => {
    const tree = renderer
        .create(<MemoryRouter><Image image='skfjlsf/psdofispdofi'/></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});