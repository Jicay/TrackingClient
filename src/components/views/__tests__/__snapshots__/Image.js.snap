// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="col align-content-center"
>
  <a
    href="/images/skfjlsf-psdofispdofi"
    onClick={[Function]}
  >
    <img
      className="img-fluid"
      src="https://live.staticflickr.com/skfjlsf/psdofispdofi.jpg"
    />
  </a>
</div>
`;
